File: lib/ts/src/internal/schema/CreateMockTelepactSchemaFromFileJsonMap.ts
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

import { MockTelepactSchema } from '../../MockTelepactSchema.js';
import { createTelepactSchemaFromFileJsonMap } from './CreateTelepactSchemaFromFileJsonMap.js';
import { getMockTelepactJson } from './GetMockTelepactJson.js';
import { copyDocumentLocators } from './DocumentLocators.js';

export function createMockTelepactSchemaFromFileJsonMap(jsonDocuments: Record<string, string>): MockTelepactSchema {
    const finalJsonDocuments = { ...jsonDocuments };
    copyDocumentLocators(jsonDocuments, finalJsonDocuments);
    finalJsonDocuments['mock_'] = getMockTelepactJson();

    const telepactSchema = createTelepactSchemaFromFileJsonMap(finalJsonDocuments);

    return new MockTelepactSchema(
        telepactSchema.original,
        telepactSchema.full,
        telepactSchema.parsed,
        telepactSchema.parsedRequestHeaders,
        telepactSchema.parsedResponseHeaders,
    );
}

File: lib/go/internal/schema/CreateMockTelepactSchemaFromFileJsonMap.go
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package schema

// CreateMockTelepactSchemaFromFileJSONMap constructs a mock Telepact schema from the supplied JSON documents map.
func CreateMockTelepactSchemaFromFileJSONMap(jsonDocuments map[string]string) (*ParsedSchemaResult, error) {
	finalDocuments := make(map[string]string, len(jsonDocuments)+1)
	for key, value := range jsonDocuments {
		finalDocuments[key] = value
	}
	CopyDocumentLocators(jsonDocuments, finalDocuments)

	finalDocuments["mock_"] = GetMockTelepactJSON()

	return CreateTelepactSchemaFromFileJSONMap(finalDocuments)
}
